src/app/register/page.tsx

```diff
@@ -0,0 +1,85 @@
+'use client';
+import React, { useState } from 'react';
+import { useRouter } from 'next/navigation';
+
+import { api } from '../api/client';
+
+const Registration = () => {
+  const [formData, setFormData] = useState({
+    name: '',
+    email: '',
+    password: '',
+  });
+
+  const [errorMessage, setError] = useState('');
+
+  const router = useRouter();
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFormData({
+      ...formData,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      const response = await api.post('user/register', {
+        json: formData,
+      });
+
+      if (response.status === 200) {
+        router.push('/login');
+      }
+    } catch (error) {
+      setError('Správně vyplňte údaje');
+    }
+  };
```

Comment:
Very similar func as before yet quite differently implemented. Maybe we should go for the same convention